test/ApplyAt.lean

```diff
@@ -0,0 +1,71 @@
+import Mathlib.Tactic.ApplyAt
+import Mathlib.Algebra.Group.Basic
+import Mathlib.Data.Nat.Basic
+import Std.Tactic.GuardExpr
+
+example {α β : Type*} (f : α → β) (a : α) : β := by
+  apply f at a
+  guard_hyp a :ₛ β
+  exact a
+
```

Comment:
```suggestion

/-- `apply at` cannot clear mvarid if still used. -/
example {α : Type} (γ : α → Type) (a : α) (f : α → γ a) : γ a := by
  apply f at a
  rename_i a₂
  guard_hyp a :ₛ γ a₂
  exact a

```

Reply:
(maybe not useful, but I wrote that trying to understand what `mainGoal.tryClear` does.)

Reply:
I think it's useful. Adding it now.